File: 0605-can-place-flowers/0605-can-place-flowers.java
class Solution {
    public boolean canPlaceFlowers(int[] flowerbed, int n) {
        int l = flowerbed.length;
        if(l == 1 && flowerbed[0] == 0 && n == 1) {
            return true;
        }

        if(l == 1 && flowerbed[0] == 1 && n > 0) {
            return false;
        }

        if(l == 1 && n >= 0) {
            return true;
        }

        for(int i=0; i<l; i++) {
            if(i == 0 && flowerbed[0] == 0 && flowerbed[1] == 0 && n > 0) {
                flowerbed[0] = 1;
                n -= 1;
            } else if(i == l-1 && flowerbed[l-1] == 0 && flowerbed[l-2] == 0 && n > 0) {
                flowerbed[l-1] = 1;
                n -= 1;
            } else if(i - 1 >= 0 && i + 1 < l && flowerbed[i] == 0 && flowerbed[i-1] == 0 && flowerbed[i+1] == 0 && n > 0) {
                flowerbed[i] = 1;
                n -= 1;
            }
        }

        return n == 0;
    }
}

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, a: List[int], n: int) -> bool:

        if len(a)==1 and a[0]==1 and n>0:
            return False
    
        if len(a)==1 and a[0]==0 and n==1:
            return True

        if len(a)==1 and a[0]==1 and n==0:
            return True

        if len(a)==1 and a[0]==0 and n==0:
            return True 
        if len(a)==1 and n>=0:
            return False
    
        for i in range(len(a)):
            if(i==0 and a[0]==0 and a[1]==0 and n>0):
                a[0]=1
                n=n-1

            elif(i==len(a)-1 and a[len(a)-1]==0 and a[len(a)-2]==0 and n>0):
                a[len(a)-1]=1
                n=n-1    
    
            elif(i+1<=len(a)-1 and i-1>=0 and a[i]==0 and a[i-1]==0 and a[i+1]==0 and n>0):
                a[i]=1
                n=n-1
    
        if n==0:
            return True
        else:
            return False
    